pkg/pgmodel/model/batch.go

```diff
@@ -68,3 +70,23 @@ func (t *Batch) Visitor() *batchVisitor {
 func (t *Batch) Absorb(other Batch) {
 	t.AppendSlice(other.data)
 }
+
+func (t *Batch) Len() int {
+	return t.CountSeries()
+}
+
+func (t *Batch) Swap(i, j int) {
+	t.data[i], t.data[j] = t.data[j], t.data[i]
+}
+
+func (t *Batch) Less(i, j int) bool {
+	s1, _, err := t.data[i].Series().GetSeriesID()
+	if err != nil {
+		log.Warn("seriesID", "not set but being sorted on")
```

Comment:
These are not recoverable errors in the core logic. These should be panic not logging.

Reply:
I was on the fence for this one. Strictly speaking the application can continue to operate even if some of seriesIDs is not being set (meaning it is zero). So I am not entirely sure if we want to crash the app. I don't feel strong either way so let me know.

Reply:
Another take is that if you would try to sort batch elsewhere you might not want sorting to crash the app.